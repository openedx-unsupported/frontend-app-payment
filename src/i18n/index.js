import { messages as paragonMessages } from '@edx/paragon';
import { messages as headerMessages } from '@edx/frontend-component-header';
import { messages as footerMessages } from '@edx/frontend-component-footer';

import arMessages from './messages/ar.json';
import frMessages from './messages/fr.json';
import es419Messages from './messages/es_419.json';
import es419MessagesProtected from '../subscription/i18n-protected-messages/es_419.json';
import zhcnMessages from './messages/zh_CN.json';
import ptMessages from './messages/pt.json';
import itMessages from './messages/it.json';
import ukMessages from './messages/uk.json';
import deMessages from './messages/de.json';
import ruMessages from './messages/ru.json';
import hiMessages from './messages/hi.json';
import frCAMessages from './messages/fr_CA.json';
import deDEMessages from './messages/de_DE.json';
import itITMessages from './messages/it_IT.json';
import ptPTMessages from './messages/pt_PT.json';
// no need to import en messages-- they are in the defaultMessage field

const appMessages = {
  ar: arMessages,
  // override the protected translations
  'es-419': { ...es419Messages, ...es419MessagesProtected },
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
  'de-de': deDEMessages,
  'it-it': itITMessages,
  'pt-pt': ptPTMessages,
};

export default [
  headerMessages,
  footerMessages,
  paragonMessages,
  appMessages,
];
