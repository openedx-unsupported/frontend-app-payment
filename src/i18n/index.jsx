import arMessages from './messages/ar.json';
import frMessages from './messages/fr.json';
import es419Messages from './messages/es_419.json';
import es419MessagesProtected from '../subscription/i18n-protected-messages/es_419.json';
import zhcnMessages from './messages/zh_CN.json';
import ptMessages from './messages/pt.json';
import itMessages from './messages/it.json';
import ukMessages from './messages/uk.json';
import deMessages from './messages/de.json';
import ruMessages from './messages/ru.json';
import hiMessages from './messages/hi.json';
import frCAMessages from './messages/fr_CA.json';
import deDEMessages from './messages/de_DE.json';
import itITMessages from './messages/it_IT.json';
import ptPTMessages from './messages/pt_PT.json';
import faMessages from './messages/fa.json';
import faIRMessages from './messages/fa_IR.json';
// no need to import en messages-- they are in the defaultMessage field

const messages = {
  ar: arMessages,
  // override the protected translations
  'es-419': { ...es419Messages, ...es419MessagesProtected },
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
  'de-de': deDEMessages,
  'it-it': itITMessages,
  'pt-pt': ptPTMessages,
  fa: faMessages,
  'fa-ir': faIRMessages,
};

export default messages;
