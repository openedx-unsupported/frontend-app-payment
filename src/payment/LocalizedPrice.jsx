import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { FormattedNumber } from '@edx/frontend-i18n';

import { localizedCurrencySelector } from './data/selectors';

/**
 * Displays a positive or negative price, according to the currency and the conversion rate set
 * in the edx-price-l10n cookie, or USD if no cookie exists.  If the currency is not USD, the
 * price is displayed with an * alert symbol next to it.
 */
function LocalizedPrice(props) {
  if (props.amount === undefined) {
    return null;
  }

  const price = props.conversionRate * props.amount;

  const priceElement = (<FormattedNumber
    value={price}
    style="currency" // eslint-disable-line react/style-prop-object
    currency={props.currencyCode}
  />);

  if (props.showAsLocalizedCurrency) {
    return (
      <span>
        {priceElement} *
      </span>
    );
  }

  return priceElement;
}

LocalizedPrice.propTypes = {
  amount: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),
  conversionRate: PropTypes.number,
  currencyCode: PropTypes.string,
  showAsLocalizedCurrency: PropTypes.bool,
};

LocalizedPrice.defaultProps = {
  amount: undefined,
  conversionRate: 1,
  currencyCode: 'USD',
  showAsLocalizedCurrency: false,
};

export default connect(localizedCurrencySelector)(LocalizedPrice);
