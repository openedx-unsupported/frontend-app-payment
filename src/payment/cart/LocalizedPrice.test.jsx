import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import Cookies from 'universal-cookie';

import LocalizedPrice from './LocalizedPrice';

jest.mock('universal-cookie', () => {
  class MockCookies {
    static result = {};

    get() {
      return MockCookies.result;
    }
  }
  return MockCookies;
});

jest.mock('@edx/frontend-platform/logging', () => ({
  logError: jest.fn(),
}));

const mockStore = configureMockStore();

describe('LocalizedPrice', () => {
  let state;

  beforeEach(() => {
    state = {
      userAccount: { email: 'person@example.com' },
      payment: {
        basket: {
          loaded: false,
          loading: false,
          products: [],
        },
      },
      i18n: {
        locale: 'en',
      },
    };
    Cookies.result = undefined;
  });

  it('should render nothing by default', () => {
    const component = (
      <IntlProvider locale="en">
        <Provider
          store={mockStore(state)}
        >
          <LocalizedPrice />
        </Provider>
      </IntlProvider>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toBeNull();
  });

  it('should render unlocalized currency', () => {
    const component = (
      <IntlProvider locale="en">
        <Provider
          store={mockStore(state)}
        >
          <LocalizedPrice amount={10} />
        </Provider>
      </IntlProvider>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render localized currency', () => {
    Cookies.result = {
      code: 'MXN',
      rate: 19,
    };
    const component = (
      <IntlProvider locale="en">
        <Provider
          store={mockStore({ ...state })}
        >
          <LocalizedPrice amount={10} />
        </Provider>
      </IntlProvider>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
