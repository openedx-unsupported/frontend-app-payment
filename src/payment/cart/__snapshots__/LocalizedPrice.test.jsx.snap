// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalizedPrice should render localized currency 1`] = `
<span>
  MX$190
   *
</span>
`;

exports[`LocalizedPrice should render unlocalized currency 1`] = `"$10.00"`;
